code/datums/gamemodes/spy_theft.dm

```diff
@@ -607,49 +624,41 @@
 
 
 	//Add station item bounties
-	var/list/S = station_bounties.Copy()
-	var/choice = 0
-	var/obj/item/item_existing = 0
-
-
-	for(var/i=1, i<=station_bounty_amt, i++)
-		choice = 0
-		difficulty = 0
-		item_existing = 0
+	var/item_choice = null
+	difficulty = 0
+	var/obj/item_existing = null
+	var/item_picked=1
+	while(item_picked<=station_bounty_amt)
+		if (station_bounties.len <= 0)
+			logTheThing( "debug", src, null, "spy_theft.dm was unable to create enough item bounties." )
+			message_admins("Spy bounty logic was unable to create enough item bounties.")
+			break
+		// Pick a known valid item, retrieve difficulty rating from other list
+		item_choice = pick(station_bounties)
+		item_existing = pick(valid_spy_thief_targets_by_type[item_choice])
```

Comment:
Ditto.

Reply:
I think I did what you were looking for.